plugins/modules/sns_topic_info.py

```diff
@@ -0,0 +1,151 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# Copyright: Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+DOCUMENTATION = r'''
+module: sns_topic_info
+short_description: sns_topic_info module
+version_added: 3.2.0
+description:
+  - The M(community.aws.sns_topic_info) module allows to get all AWS SNS topics or properties of a specific AWS SNS topic.
+author:
+- "Alina Buzachis (@alinabuzachis)"
+options:
+  topic_arn:
+    description:
+      - The ARN of the AWS SNS topic for which you wish to find subscriptions or list attributes.
+    required: false
+    type: str
+extends_documentation_fragment:
+- amazon.aws.aws
+- amazon.aws.ec2
+'''
+
+EXAMPLES = r'''
+- name: list all the topics
+  community.aws.sns_topic_info:
+  register: sns_topic_list
+
+- name: get info on specific topic
+  community.aws.sns_topic_info:
+    topic_arn: "{{ sns_arn }}"
+  register: sns_topic_info
+'''
+
+RETURN = r'''
+changed:
+  description: True if listing the AWS SNS topics succeeds
+  type: bool
+  returned: always
+  sample: false
+result:
+  description:
+    - The result contaning the details of one or all AWS SNS topics.
+  returned: suceess
+  type: list
+  contains:
+    sns_arn:
+        description: The ARN of the topic.
+        type: str
+        returned: always
+        sample: "arn:aws:sns:us-east-2:111111111111:my_topic_name"
+    sns_topic:
+        description: Dict of sns topic details
+        type: complex
+        returned: always
+        contains:
+            delivery_policy:
+                description: Delivery policy for the SNS topic
+                returned: when topic is owned by this AWS account
+                type: str
```

Comment:
```suggestion
                type: dict
```

Reply:
It should be string (to be consistent with sns_topic), I corrected the sample. Thank you for catching this. Same for `delivery_policy`.